src/gather_classes.rs

```diff
@@ -257,80 +255,131 @@ fn update_report(
+                if string_is_same(string) || string_is_empty(string) {
+                    Ok(())
+                } else {
+                    Err(anyhow!("unexpected string: {}", string))
+                }
             }
-        }
-        Value::Array(call_list) => {
-            let nested_calls = call_list
-                .iter()
-                .map(|inner_call| get_calls(inner_call))
-                .collect::<Result<Vec<_>, _>>()?
-                .into_iter()
-                .flatten();
-            Ok(Box::new(call_list.iter().chain(nested_calls)))
-        }
-        Value::Object(obj_map) => {
-            if let Some(Value::Object(diff_map)) = obj_map.get(DIFFERENT) {
-                let base_list = diff_map
-                    .get("base")
-                    .ok_or(anyhow!("difference should have base"))?;
-                let native_list = diff_map
-                    .get("native")
-                    .ok_or(anyhow!("difference should have native"))?;
-                // TODO (#72) Compare and get diffs. For now we just chain.
-                Ok(Box::new(
-                    get_calls(base_list)?.chain(get_calls(native_list)?),
-                ))
-            } else if let Some(calls_value) = obj_map.get("calls") {
-                get_calls(calls_value)
-            } else if let Some(_val) = obj_map.get("revert_reason") {
-                Ok(Box::new(::std::iter::empty()))
-            } else {
-                Err(anyhow!("unexpected object: `{}`", obj))
+            Value::Array(call_list) => {
+                for call in call_list {
+                    get_calls_inner(call, result)?;
+                }
+                Ok(())
+            }
+            Value::Object(obj_map) => {
+                // If call is different, then base and native will have their own list of calls
+                if let Some(Value::Object(diff_map)) = obj_map.get(DIFFERENT) {
+                    let base_list = diff_map
+                        .get("base")
+                        .ok_or(anyhow!("difference should have base"))?;
+                    let native_list = diff_map
+                        .get("native")
+                        .ok_or(anyhow!("difference should have native"))?;
+
+                    let mut base_calls = Vec::new();
+                    get_calls_inner(base_list, &mut base_calls)?;
+                    let mut native_calls = Vec::new();
+                    get_calls_inner(native_list, &mut native_calls)?;
+
+                    result.extend(merge_calls_with_count(base_calls, native_calls));
+                } else {
+                    match get_call_with_count(obj) {
+                        Ok(current_call) => {
+                            result.push(current_call);
+                        }
+                        // We let the parsing fail silently, as there may be inner calls that are valid
```

Comment:
Could you expand on this explanation?

Reply:
Updated the comment with Pearson's description